src/spyglass/decoding/v0/dj_decoder_conversion.py

```diff
@@ -0,0 +1,201 @@
+"""Converts decoder classes into dictionaries and dictionaries into classes
+so that datajoint can store them in tables."""
+from spyglass.utils import logger
+
+try:
+    from replay_trajectory_classification.continuous_state_transitions import (
+        Identity,
+        RandomWalk,
+        RandomWalkDirection1,
+        RandomWalkDirection2,
+        Uniform,
+    )
+    from replay_trajectory_classification.discrete_state_transitions import (
+        DiagonalDiscrete,
+        RandomDiscrete,
+        UniformDiscrete,
+        UserDefinedDiscrete,
+    )
+    from replay_trajectory_classification.environments import Environment
+    from replay_trajectory_classification.initial_conditions import (
+        UniformInitialConditions,
+        UniformOneEnvironmentInitialConditions,
+    )
+    from replay_trajectory_classification.observation_model import (
+        ObservationModel,
+    )
+    from track_linearization import make_track_graph
+
+    def _convert_dict_to_class(d: dict, class_conversion: dict) -> object:
```

Comment:
These funcs seem like good candidate for a more object oriented approach. By turning them into a class, I think you could avoid the `TypeError` complications below and cut down on the embedded dict calls (eg `a[b][c]`)